File: src/Zamka/Zamka.java
import java.io.*;

public class Zamka {
    public static void main(String[] args) throws IOException {
        InputStreamReader inp = new InputStreamReader(System.in);
        BufferedReader sc = new BufferedReader(inp);
        PrintWriter writer = new PrintWriter(System.out);
        
        int l = Integer.parseInt(sc.readLine());
        int d = Integer.parseInt(sc.readLine());
        int x = Integer.parseInt(sc.readLine());
        int minSum = 10001;
        int maxSum = 0;
        for (int i = l; i <= d; i++) {
            int sumDigit = 0;
            int temp = i;
            while (temp > 0) {
                sumDigit += temp % 10;
                temp /= 10;
            }
            if (sumDigit == x && i > maxSum) {
                maxSum = i;
            }
            if (sumDigit == x && i < minSum) {
                minSum = i;
            }
        }

        writer.println(minSum);
        writer.println(maxSum);
        writer.flush();
    }
}

File: src/Zamka/zamka.cpp
#include <iostream>
using namespace std;

int main()
{
    int l, d, x, temp, sd;
    int minsum = 10001;
    int maxsum = 0;
    cin >> l >> d >> x;
    for (int i = l; i <= d; i++) {
        temp = i;
        sd = 0;
        while (temp > 0) {
            sd += temp % 10;
            temp /= 10;
        }
        if (sd == x && i > maxsum) {
            maxsum = i;
        }
        if (sd == x && i < minsum) {
            minsum = i;
        }
    }

    cout << minsum << "\n" << maxsum;
    return 0;
}